pkg/helm/chartproxy/proxy.go

```diff
@@ -99,10 +104,20 @@ func (p *proxy) IndexFile(onlyCompatible bool) (*repo.IndexFile, error) {
 					}
 				}
 				if len(entry) > 0 {
+					if overwrites != "" {
+						klog.Infof("%s overwrites repo %s", helmRepo.Name, overwrites)
+						delKeys = append(delKeys, entry[0].Name+"--"+overwrites)
+					}
+
 					indexFile.Entries[key+"--"+helmRepo.Name] = entry
 				}
 			}
 		}
 	}
+
+	for _, delKey := range delKeys {
+		klog.Infof("Deleting helm index key: %s", delKey)
```

Comment:
not sure if we want to create so many log messages?

Reply:
OK

Reply:
Removed